src/panels/config/integrations/integration-panels/zha/zha-network-visualization-page.ts

```diff
@@ -260,16 +266,12 @@ export class ZHANetworkVisualizationPage extends LitElement {
               dashes: neighbor.relationship !== "Child",
             });
           } else {
-            edges[idx].color = this._getLQI(
-              (parseInt(edges[idx].label!) + parseInt(neighbor.lqi)) / 2
-            ).color;
-            edges[idx].width = this._getLQI(
-              (parseInt(edges[idx].label!) + parseInt(neighbor.lqi)) / 2
-            ).width;
-            edges[idx].length =
-              2000 -
-              6 * ((parseInt(edges[idx].label!) + parseInt(neighbor.lqi)) / 2);
-            edges[idx].label += "/" + neighbor.lqi;
+            const effective_lqi = Math.min(parseInt(edges[idx].label!), parseInt(neighbor.lqi));
+            edges[idx].label += " & " + neighbor.lqi;
+            edges[idx].color = this._getEdgeOptions(effective_lqi).color;
+            edges[idx].width = this._getEdgeOptions(effective_lqi).width;
+            edges[idx].length = this._getEdgeOptions(effective_lqi).length;
+            edges[idx].physics = this._getEdgeOptions(effective_lqi).physics;
```

Comment:
Add a const edge_options to calculate the edge options beforehand. The method is getting lengthy and we're requesting it already four times here. (Same for the neighbor structure above)

Reply:
Addressed in https://github.com/home-assistant/frontend/pull/24327/commits/10232a1fb57d61001ab54e8f10032aa5fdf491db